RadixWallet/Prelude/Extensions/IdentifiedArrayOf+Extensions.swift

```diff
@@ -5,3 +5,44 @@ extension IdentifiedArrayOf {
 		return copy
 	}
 }
+
+extension Collection {
+	public func identifiablyEnumerated() -> [OffsetIdentified<Element>] {
+		enumerated().map(OffsetIdentified.init)
+	}
+}
+
+extension Collection where Element: Identifiable {
+	public func identified() throws -> IdentifiedArrayOf<Element> {
+		guard Set(map(\.id)).count == count else {
+			throw IdentifiedArrayError.clashingIDs
+		}
+		return .init(uniqueElements: self)
+	}
+
+	public func uniqueIdentified() -> IdentifiedArrayOf<Element> {
+		.init(uncheckedUniqueElements: self)
```

Comment:
this crashes for duplicates... we ought to never crash in PROD. Best to never use `unchecked`, it has bit us in the a** in the past. hmm is it used? can we filter out copies in PROD instead (but crash in DEBUG?)?